packages/api/src/command/medical/cq-directory/parse-cq-directory-entry.ts

```diff
@@ -77,16 +79,17 @@ function getUrls(contained: Contained): XCUrls | undefined {
   return urls;
 }
 
-function getCoordinates(address: Address[] | undefined): { lat: string; lon: string } | undefined {
+function getCoordinates(address: Address[] | undefined): { lat: number; lon: number } | undefined {
   if (!address) return;
   const coordinates = address.flatMap(a => {
     if (a.extension?.url === ORG_POSITION) {
       const position = a.extension?.valueCodeableConcept?.coding?.value?.position;
       if (!position) return [];
-      return {
-        lat: position.latitude.value,
-        lon: position.longitude.value,
-      };
+      const lat = parseFloat(position.latitude.value);
+      const lon = parseFloat(position.longitude.value);
+      if (lat < -90 || lat > 90) return [];
+      if (lon < -180 || lon > 180) return [];
```

Comment:
parseFloat can return NaN if non numeric characters in position lat long. Maybe check for NaNs:
`if (isNaN(lat) || isNaN(lon)) return []; // Check if lat or lon are NaN
`

Reply:
oh cool, didnt know that!